File: fundref_test.ts
//
// Test our FundRef ID methods
//
import { assertEquals, assertNotEquals } from "@std/assert";

import { normalizeFundRefID, validateFundRefID } from "./fundref.ts";
import { verifyFundRefID } from "./fundref_record.ts";

const varified_ids: string[] = [
  "10.13039/100000001",
  "10.13039/100000009",
];

for (const id of varified_ids) {
  // Normalize
  const normalized = normalizeFundRefID(id);
  assertNotEquals(normalized, undefined);
  console.log(`Normalized FundRef ID: ${normalized}`);
  // Validate
  assertEquals(validateFundRefID(id), true);
  // Verify is via CrossRef
  const isOK: boolean = await verifyFundRefID(id);
  assertEquals(isOK, true);
}

File: fundref_test.go
package metadatatools

import "testing"

func TestFundRefID(t *testing.T) {
	if testing.Short() {
		t.Skip("skipping network test in short mode")
	}

	verifiedIDs := []string{
		"10.13039/100000001",
		"10.13039/100000009",
	}

	for _, id := range verifiedIDs {
		normalized := NormalizeFundRefID(id)
		if normalized == "" {
			t.Errorf("NormalizeFundRefID(%q) returned an empty string", id)
		}
		t.Logf("Normalized FundRef ID: %s", normalized)

		if !ValidateFundRefID(id) {
			t.Errorf("ValidateFundRefID(%q) = false, expected true", id)
		}

		if !VerifyFundRefID(id) {
			t.Errorf("VerifyFundRefID(%q) = false, expected true", id)
		}
	}
}
